Mimir.Worker/BlockPoller.cs

```diff
@@ -60,25 +72,104 @@ public async Task RunAsync(CancellationToken cancellationToken)
+            var avatarDataArray = await Task.WhenAll(
+                avatarAddresses.Select(stateGetter.GetAvatarData)
+            );
             await mongoDbStore.BulkUpsertAvatarDataAsync(
-                avatarDataArray
-                    .Where(e => e is not null)
-                    .OfType<AvatarData>()
-                    .ToList());
+                avatarDataArray.Where(e => e is not null).OfType<AvatarData>().ToList()
+            );
+        }
+    }
+
+    private async Task EveryPatchTableAsync(
+        long processBlockIndex,
+        CancellationToken cancellationToken
+    )
+    {
+        var rawPatchTableTxsResp = await headlessGqlClient.GetPatchTableTransactions.ExecuteAsync(
+            processBlockIndex,
+            cancellationToken
+        );
+        if (rawPatchTableTxsResp.Data is null)
+        {
+            HandleErrors(rawPatchTableTxsResp);
+            return;
+        }
+
+        var patchTableTxs = rawPatchTableTxsResp
+            .Data?.Transaction?.NcTransactions.Where(raw => raw is not null)
+            .Select(raw =>
+                TxMarshaler.DeserializeTransactionWithoutVerification(
+                    Convert.FromBase64String(raw!.SerializedPayload)
+                )
+            )
+            .ToList();
+        foreach (var patchTableTx in patchTableTxs)
+        {
+            var patchTableAction = (Dictionary)patchTableTx.Actions[0];
+            var patchTableActionValues = (Dictionary)patchTableAction["values"];
+            var tableName = ((Text)patchTableActionValues["table_name"]).ToDotnetString();
+
+            var sheetType = typeof(ISheet)
+                .Assembly.GetTypes()
+                .Where(type =>
+                    type.Namespace is { } @namespace
+                    && @namespace.StartsWith($"{nameof(Nekoyume)}.{nameof(Nekoyume.TableData)}")
+                    && !type.IsAbstract
+                    && typeof(ISheet).IsAssignableFrom(type)
+                    && type.Name == tableName
+                )
+                .FirstOrDefault();
+
+            if (sheetType == null)
+            {
+                throw new TypeLoadException(
+                    $"Unable to find a class type matching the table name '{tableName}' in the specified namespace."
+                );
+            }
+
+            var sheetAddress = Addresses.TableSheet.Derive(tableName);
+            var sheetState = await stateService.GetState(sheetAddress);
+            if (sheetState is not Text sheetValue)
+            {
+                throw new InvalidOperationException(
+                    $"Expected sheet state to be of type 'Text'."
+                );
+            }
+
+            var sheetInstance = Activator.CreateInstance(sheetType);
+            if (sheetInstance is not ISheet sheet)
+            {
+                throw new InvalidCastException($"Type {sheetType.Name} cannot be cast to ISheet.");
+            }
```

Comment:
It would be easier to read if we created the `sheetInstance` immediately after inspecting the `sheetType` variable, as shown below.

```suggestion
            if (sheetType == null)
            {
                throw new TypeLoadException(
                    $"Unable to find a class type matching the table name '{tableName}' in the specified namespace."
                );
            }

            var sheetInstance = Activator.CreateInstance(sheetType);
            if (sheetInstance is not ISheet sheet)
            {
                throw new InvalidCastException($"Type {sheetType.Name} cannot be cast to ISheet.");
            }

            var sheetAddress = Addresses.TableSheet.Derive(tableName);
            var sheetState = await stateService.GetState(sheetAddress);
            if (sheetState is not Text sheetValue)
            {
                throw new InvalidOperationException(
                    $"Expected sheet state to be of type 'Text'."
                );
            }
```

Reply:
Good, I changed the order of the code.